src/ralph/backends/data/async_ws.py

```diff
@@ -0,0 +1,228 @@
+            Setting it to `None` removes the header.
+        write_limit (int): High-water mark of write buffer in bytes.
+    """
+
+    close_timeout: Optional[float] = None
+    compression: Optional[str] = "deflate"
+    max_size: Optional[int] = 2**20
+    max_queue: Optional[int] = 2**5
+    open_timeout: Optional[float] = 10
+    origin: Optional[str] = None
+    ping_interval: Optional[float] = 20
+    ping_timeout: Optional[float] = 20
+    read_limit: int = 2**16
+    user_agent_header: Optional[str] = USER_AGENT
+    write_limit: int = 2**16
+
+
+class WSDataBackendSettings(BaseDataBackendSettings):
+    """Websocket data backend default configuration.
+
+    Attributes:
+        CLIENT_OPTIONS (dict): A dictionary of valid options for the websocket client
+            connection. See `WSClientOptions`.
+        URI (str): The URI to connect to.
+    """
+
+    class Config(BaseSettingsConfig):
+        """Pydantic Configuration."""
+
+        env_prefix = "RALPH_BACKENDS__DATA__WS__"
+
+    CLIENT_OPTIONS: WSClientOptions = WSClientOptions()
+    URI: AnyUrl
+
+
+class AsyncWSDataBackend(BaseAsyncDataBackend[WSDataBackendSettings, BaseQuery]):
+    """Websocket stream backend."""
+
+    name = "async_ws"
+
+    def __init__(self, settings: Optional[WSDataBackendSettings] = None):
+        """Instantiate the async websocket data backend.
+
+        Args:
+            settings (WSDataBackendSettings or None): The websocket backend settings.
+                If `settings` is `None`, a default settings instance is used instead.
+        """
+        super().__init__(settings)
+        self._client = None
+
+    async def client(self) -> websockets.WebSocketClientProtocol:
+        """Create a websocket client connected to `settings.URI` if it doesn't exist."""
+        if not self._client:
+            try:
+                self._client = await websockets.connect(
+                    self.settings.URI, **self.settings.CLIENT_OPTIONS.dict()
+                )
+            except (websockets.WebSocketException, OSError, TimeoutError) as error:
+                msg = "Failed open websocket connection for %s: %s"
+                self.logger.error(msg, self.settings.URI, error)
+                raise BackendException(msg % (self.settings.URI, error)) from error
+        return self._client
+
+    async def status(self) -> DataBackendStatus:
+        """Implement data backend checks (e.g. connection, cluster status).
+
+        Return:
+            DataBackendStatus: The status of the data backend.
+        """
+        try:
+            client = await self.client()
+            pong_waiter = await client.ping()
+            await pong_waiter
+        except BackendException:
+            return DataBackendStatus.ERROR
+        except (websockets.WebSocketException, RuntimeError) as error:
+            self.logger.error("Failed to Ping %s: %s", self.settings.URI, error)
+            return DataBackendStatus.ERROR
```

Comment:
Maybe `AWAY` would be more appropriate?

```suggestion
            return DataBackendStatus.AWAY
```

Reply:
Good point) Thanks) Updated it)